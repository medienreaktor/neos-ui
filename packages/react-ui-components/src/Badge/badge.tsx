import mergeClassNames from 'classnames';
import React, {PureComponent} from 'react';

interface BadgeTheme {
    readonly badge: string;
}

export interface BadgeProps {
    /**
     * An optional `className` to attach to the wrapper.
     */
    readonly className?: string;

    /**
     * Badge's label.
     */
    readonly label: string;

    /**
     * An optional css theme to be injected.
     */
    readonly theme?: BadgeTheme;
}

class Badge extends PureComponent<BadgeProps> {
    public render(): JSX.Element {
        const {
            className,
            theme,
            label,
            ...rest
        } = this.props;
        const finalClassName = mergeClassNames(theme!.badge, className);

        return (
            <div {...rest} className={finalClassName}>{label}</div>
        );
    }
}

export default Badge;
