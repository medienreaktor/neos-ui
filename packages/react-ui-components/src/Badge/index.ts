import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from '../identifiers';

import Badge from './badge';
import style from './style.css';

export default themr(identifiers.badge, style)(Badge);
