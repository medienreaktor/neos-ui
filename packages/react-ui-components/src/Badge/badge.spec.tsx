import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import Badge, {BadgeProps} from './badge';

describe('<Badge/>', () => {
    const props: BadgeProps = {
        label: 'Foo children',
        theme: {
            badge: 'badgeClassName',
        },
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<Badge {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<Badge {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });
});
