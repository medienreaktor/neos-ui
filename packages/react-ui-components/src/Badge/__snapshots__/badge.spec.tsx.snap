// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> should render correctly. 1`] = `
<div
  className="badgeClassName"
>
  Foo children
</div>
`;
